From 97a7097b98fab4d6ad25a7e7e7af67492c147199 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 16:48:00 +0200
Subject: [PORT FROM MAIN] Telephony: TZ update when the country code is NOK

BZ: 33279

3GPP Conformance 26.7.6.1.1 is failing because the telephony
framework doesn't update NITZ timezeone correctly.

Steps to Reproduce: Execute 3GPP Conformance 26.7.6.1.1
1. UE receives MM INFORMATION containing "0 hour" as a timezone.
Expected Results: UE updates local timezone with GMT+0 value.
Actual Results: UE doesn't update local timezone and keeps
preceding value.

This patch integrates the TZ update when the check TZ is not OK
(Country Code NOK)

Change-Id: I231fefaec52f31d4d6ecb05564e1805b13854d61
Orig-Change-Id: I1f697fc8924b3b1afd69952e454446962517ec7a
Signed-off-by: Emmanuel Delaude <emmanuelx.delaude@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/75821
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index d4d2edc..f865953 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -1404,6 +1404,7 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
                 mZoneOffset  = tzOffset;
                 mZoneDst     = dst != 0;
                 mZoneTime    = c.getTimeInMillis();
+                zone = getNitzTimeZone(mZoneOffset, mZoneDst, mZoneTime);
             }
 
             if (zone != null) {
-- 
1.7.4.1

